### PR TITLE
will add functionality to the PATCH endpoint to update the votes cou…

### PR DESCRIPTION
**Pull request by marklovejoydev**
…nt by sending an object given an amount to increase by also attached is the tests linked to this function and the update to endpoints

### CODE REVIEW

File: controller/article-controller.js [hunk 1]
```diff
@@ -35,10 +35,22 @@ exports.getCommentsByArticleId = (req, res, next) => {
...
+    .then(() => insertComments(username, body, article_id))
     .then((comment)=>{
         res.status(201).send(comment)
     })
     .catch(next)
+}
+
+exports.patchArticle = (req, res , next) => {
+    const votes = req.body.inc_votes
+    const id = req.params.article_id
```
Comment (Keb84) [inline controller/article-controller.js:49]: Maybe a nice destructure would look neat here instead? 👀

File: model/article-model.js [hunk 2]
```diff
@@ -64,3 +64,14 @@ exports.insertComments = (username, body, article_id) => {
...
 
+  exports.updateArticle = (votes, id) => {
+  return db.query(`
+  UPDATE articles
+  SET votes = votes + $1
+  WHERE article_id = $2
+  RETURNING *;
+  `, [votes, id])
+  .then(({rows})=> {
+    return rows[0]
```
Comment (Keb84) [inline model/article-model.js:75]: Do you think an custom error would be needed in here if the article ID isn't in the database?

File: controller/article-controller.js [hunk 3]
```diff
@@ -35,10 +35,22 @@ exports.getCommentsByArticleId = (req, res, next) => {
...
+    checkArticleExists(id)
```
Comment (Keb84) [inline controller/article-controller.js:50]: You shouldn't need to check if the article ID exists here. We should allow psql to throw an error in the model then we can make the check and use a custom error in there.

**Review by Keb84 — CHANGES_REQUESTED** [review]
Good work on this ticket 🤸 please check out my comments and resubmit 🎸

**Review by RichardNeat — APPROVED** [review]
Nice one Mark! Merge away :)

**Pull request merged**